src/Evolution/Systems/GeneralizedHarmonic/DampedHarmonicGaugeQuantities.hpp

```diff
@@ -0,0 +1,354 @@
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class DataVector;  // IWYU pragma: keep
+/// \endcond
+
+// IWYU pragma: no_forward_declare Tags::Time
+// IWYU pragma: no_forward_declare Tags::Coordinates
+// IWYU pragma: no_forward_declare Tags::deriv
+// IWYU pragma: no_forward_declare Tensor
+// IWYU pragma: no_forward_declare db::ComputeTag
+// IWYU pragma: no_forward_declare gr::InverseSpatialMetric
+// IWYU pragma: no_forward_declare gr::Tags::Lapse
+// IWYU pragma: no_forward_declare gr::Tags::Shift
+// IWYU pragma: no_forward_declare gr::Tags::SpacetimeMetric
+// IWYU pragma: no_forward_declare gr::Tags::SpacetimeNormalOneForm
+// IWYU pragma: no_forward_declare gr::Tags::SpacetimeNormalVector
+// IWYU pragma: no_forward_declare gr::Tags::InverseSpatialMetric
+// IWYU pragma: no_forward_declare gr::Tags::SqrtDetSpatialMetric
+
+namespace GeneralizedHarmonic {
+namespace DampedHarmonicGaugeHelpers {
+/*!
+ * \ingroup GeneralizedHarmonic
+ * \brief Compute the spatial weight functions used in the damped
+ *        harmonic gauge source function.
+ *
+ * \details The spatial weight function is:
+ * \f[ W(x^i) = \mathrm{exp}(- \sigma_x (r / r_{\mathrm{max}})^2) \f]
+ */
+template <size_t SpatialDim, typename Frame, typename DataType>
+Scalar<DataType> weight_function(
+    const tnsr::I<DataType, SpatialDim, Frame>& coords,
+    const double& r_max, const double& r_width) noexcept;
+
+/*!
+ * \ingroup GeneralizedHarmonic
+ * \brief Computes the spatetime derivative of the weight function in H
+ *
+ * \details Compute the derivative:
+ * \f{align*}
+ * \partial_a W(x^i)=& \partial_a \mathrm{exp}(- \sigma_x(r/r_\mathrm{max})^2 \\
+ *                  =& (-2 * x^i * \sigma_x / r_\mathrm{max}^2) *
+ *                        exp[-\sigma_x (r/r_\mathrm{max})^2
+ * \f}
+ */
+template <size_t SpatialDim, typename Frame, typename DataType>
+tnsr::a<DataType, SpatialDim, Frame> spacetime_deriv_of_weight_function(
+    const tnsr::I<DataType, SpatialDim, Frame>& coords,
+    const double& r_max, const double& r_width) noexcept;
+
+/*!
+ * \ingroup GeneralizedHarmonic
+ * \brief Computes the roll-on function for the damped harmonic gauge
+ *
+ * \details For times after \f$t_0\f$, compute the roll-on function
+ * \f$ R(t) = 1 - \mathrm{exp}(-((t - t_0)/\sigma_t)^4]) \f$,
+ * and return \f$ R(t) = 0\f$ at times before.
+ */
+double roll_on_function(const double& time, const double& t_start,
+    const double& sigma_t) noexcept;
+
+/*!
+ * \ingroup GeneralizedHarmonic
+ * \brief Computes the time derivative of the gauge roll-on function
+ *
+ * \details Compute the time derivative:
+ * \f{align*}
+ * \partial_0 R(t) = 4 exp[-((t - t0)/\sigma_t)^4] (t - t0)^3 / sigma_t^4
+ * \f}
+ */
+double time_deriv_of_roll_on_function(const double& time, const double& t_start,
+    const double& sigma_t) noexcept;
+
+/*!
+ * \ingroup GeneralizedHarmonic
+ * \brief Computes the log factor that appears in damped harmonic gauge source
```

Comment:
This comment should only refer to `log_factor`. Make an extra comment for `spacetime_deriv_of_log_factor` and `spacetime_deriv_of_power_log_factor`.

Reply:
done